### PR TITLE
tests: Bluetooth: df: migrate direction finding unit tests to new ZTEST API

### PR DESCRIPTION
**Pull request by ppryga-nordic**
Migrate tests in `connected_cte_req`, `connection_cte_tx_params`, `connectionless_cte_chains`, `connectionless_cte_rx`, `connectionless_cte_tx`  to new ZTEST API.

Remove main.c source file because is unnecessary when automatic
unit tests registration is available.

Fix issues related with Bluetooth Host and Controller re-initialization that were found during migration.
For details please see the commit messaged.

Part of: #47002

### CODE REVIEW

File: subsys/bluetooth/host/hci_core.c [hunk 1]
```diff
@@ -3736,6 +3736,10 @@ int bt_disable(void)
 	/* Clear BT_DEV_ENABLE here to prevent early bt_enable() calls */
 	atomic_clear_bit(bt_dev.flags, BT_DEV_ENABLE);
 
+#if defined(CONFIG_BT_PER_ADV_SYNC)
+	bt_periodic_sync_disable();
+#endif /* CONFIG_BT_PER_ADV_SYNC */
```
Comment (Thalley) [inline subsys/bluetooth/host/hci_core.c:3741]: It sort of feels like this should be done before `atomic_clear_bit(bt_dev.flags, BT_DEV_ENABLE);`

File: subsys/bluetooth/host/scan.c [hunk 2]
```diff
@@ -723,6 +723,13 @@ static struct bt_le_per_adv_sync *get_pending_per_adv_sync(void)
 	return NULL;
 }
 
+void bt_periodic_sync_disable(void)
+{
+	for (int i = 0; i < ARRAY_SIZE(per_adv_sync_pool); i++) {
```
Comment (Thalley) [inline subsys/bluetooth/host/scan.c:728]:
```suggestion
	for (size_t i = 0; i < ARRAY_SIZE(per_adv_sync_pool); i++) {
```

File: tests/bluetooth/df/common/include/bt_common.h [hunk 3]
```diff
@@ -4,4 +4,5 @@
  * SPDX-License-Identifier: Apache-2.0
  */
 
-void ut_bt_setup(void);
+void *ut_bt_setup(void);
```
Comment (Thalley) [inline tests/bluetooth/df/common/include/bt_common.h:7]: What is the purpose of modifying the return value if we always return `NULL`?

**Review by Thalley — CHANGES_REQUESTED** [review]

File: tests/bluetooth/df/common/include/bt_common.h (hunk 3 above)
Comment (ppryga) [reply tests/bluetooth/df/common/include/bt_common.h:7]: The change is caused by ZTEST framework, signature of a suite and unit tests setup, teardown functions has changed.

File: tests/bluetooth/df/connectionless_cte_rx/src/test_set_iq_sampling_enable.c [hunk 4]
```diff
@@ -280,62 +282,34 @@ void test_set_scan_cte_rx_disable_when_disabled(void)
...
-		ztest_unit_test(test_set_scan_cte_rx_disable_when_disabled));
-	ztest_run_test_suite(test_hci_set_scan_cte_rx_enable);
+	ut_bt_setup();
+
+	common_create_per_sync_set();
+
+	return NULL;
 }
+
+static void common_per_sync_teardown(void *data) { ut_bt_teardown(data); }
```
[the remark this answers is not in the document]
Comment (ppryga) [reply tests/bluetooth/df/connectionless_cte_rx/src/test_set_iq_sampling_enable.c:310]: Ok, I'll change that.

File: tests/bluetooth/df/common/include/bt_common.h (hunk 3 above)
Comment (Thalley) [reply tests/bluetooth/df/common/include/bt_common.h:7]: Roger that :)

File: subsys/bluetooth/host/hci_core.c (hunk 1 above)
Comment (ppryga) [reply subsys/bluetooth/host/hci_core.c:3738]: Moved up and extended comment related with clear of `BT_DEV_ENABLE` bit.

File: subsys/bluetooth/host/scan.c (hunk 2 above)
Comment (ppryga) [reply subsys/bluetooth/host/scan.c]: Changed

**Review by Thalley — DISMISSED** [review]

**Comment by ppryga** [comment]
Fixed compilation warning, by accident I've removed a line in `scan.h` file that was `bt_hci_get_per_adv_sync` declaration.

**Review by Thalley — APPROVED** [review]

**Review by carlescufi — APPROVED** [review]

**Pull request merged**